### PR TITLE
back end comment routre

### PR DESCRIPTION
**Pull request by MohammedSallout**
- GET all comments
- POST new comment
- GET comment by id
- PUT to update comment
- DELETE comment

### CODE REVIEW

File: src/controllers/comment/deleteComment.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { type Request, type Response, type NextFunction } from 'express'
+import deleteCommentQuery from '../../queries/comment/deleteComment'
+import CustomError from '../../helpers/CustomError'
+
+const deleteComment = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+  try {
+    const { commentId } = req.params
```
Comment (Mu7ammadAbed) [inline src/controllers/comment/deleteComment.ts:7]: Where's the validation for this?

File: src/controllers/comment/getComments.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { type NextFunction, type Request, type Response } from 'express'
+import getCommentsQuery from '../../queries/comment/getComments'
+
+const getComments = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+  const postId = req.params.postId
```
Comment (Mu7ammadAbed) [inline src/controllers/comment/getComments.ts:5]:
```suggestion
  const { postId } = req.params
```

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

File: src/routes/index.ts [hunk 3]
```diff
@@ -1,10 +1,12 @@
 import express from 'express'
 import authRouter from './auth'
+import commentRouter from './comment'
 import categoryRouter from './category'
 
 const router = express.Router()
 
 router.use('/auth', authRouter)
+router.use(commentRouter)
```
Comment (Mu7ammadAbed) [inline src/routes/index.ts:9]:
```suggestion
router.use('/comments', commentRouter)
```

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

File: src/routes/comment.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import addComment from '../controllers/comment/addComment'
+import deleteComment from '../controllers/comment/deleteComment'
+import updateComment from '../controllers/comment/updateComment'
+import authUser from '../middlewares/auth'
+import checkType, { userTypes } from '../middlewares/checkType'
+const commentRouter = express.Router()
+
+const { REGULAR, ADMIN } = userTypes
+
+commentRouter.get('/post/:postId/comments', getComments)
```
Comment (Mu7ammadAbed) [inline src/routes/comment.ts:13]: this shouldn't be here

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by AbdallahAbujazar — APPROVED** [review]

**Review by haitham-akram — APPROVED** [review]

**Pull request merged**